static/js/components/NewsFeed.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import dayjs from 'dayjs';
+import relativeTime from 'dayjs/plugin/relativeTime';
+
+import WidgetPrefsDialog from './WidgetPrefsDialog';
+import * as ProfileActions from '../ducks/profile';
+import styles from './NewsFeed.css';
+
+dayjs.extend(relativeTime);
+
+
+const NewsFeed = () => {
+  const { newsFeedItems } = useSelector((state) => state.newsFeed);
+  const profile = useSelector((state) => state.profile);
+
+  const newsFeedPrefs = (
+    profile != null &&
+    Object.prototype.hasOwnProperty.call(profile, "preferences") &&
+    profile.preferences != null &&
+    Object.prototype.hasOwnProperty.call(profile.preferences, "newsFeed")
+  ) ?
+    profile.preferences.newsFeed : { numItems: "" };
+  if (newsFeedItems === undefined) {
+    return (
+      <div>
+        No new items to display...
+      </div>
+    );
+  }
+  return (
+    <div style={{ border: "1px solid #DDD", padding: "10px" }}>
+      <h2 style={{ display: "inline-block" }}>
+        News Feed
+      </h2>
+      <div style={{ display: "inline-block", float: "right" }}>
+        <WidgetPrefsDialog
+          formValues={newsFeedPrefs}
+          stateBranchName="newsFeed"
+          title="News Feed Preferences"
+          onSubmit={ProfileActions.updateUserPreferences}
```

Comment:
What? We _are_ passing `onSubmit` as a prop. What do you mean by "components are not supposed to be aware of redux state at all"? We frequently access the redux state with `useSelector`. Can you please clarify?

Reply:
Yes, we previously had these sub-divided into components and containers.

Reply:
These are all of the places we access the redux state from within components: 
```
static/js/components/CommentList.jsx:  const source = useSelector((state) => state.source);
static/js/components/CommentList.jsx:  const userProfile = useSelector((state) => state.profile);
static/js/components/CommentList.jsx:  const acls = useSelector((state) => state.profile.acls);
static/js/components/Footer.jsx:  const version = useSelector((state) => state.sysInfo.version);
static/js/components/Group.jsx:  const currentUser = useSelector((state) => state.profile);
static/js/components/Group.jsx:  const group = useSelector((state) => state.group);
static/js/components/GroupList.jsx:    groups = useSelector((state) => state.groups.latest);
static/js/components/Groups.jsx:  const roles = useSelector((state) => state.profile.roles);
static/js/components/HomePage.jsx:  const groups = useSelector((state) => state.groups.latest);
static/js/components/Logo.jsx:  const rotateLogo = useSelector((state) => state.logo.rotateLogo);
static/js/components/NewsFeed.jsx:  const { newsFeedItems } = useSelector((state) => state.newsFeed);
static/js/components/NewsFeed.jsx:  const profile = useSelector((state) => state.profile);
static/js/components/Plot.jsx:  const plots = useSelector((state) => state.plots);
static/js/components/Profile.jsx:  const profile = useSelector((state) => state.profile);
static/js/components/Profile.jsx:  const groups = useSelector((state) => state.groups.latest);
static/js/components/ProfileDropdown.jsx:  const profile = useSelector((state) => state.profile);
static/js/components/Source.jsx:  const source = useSelector((state) => state.source);
static/js/components/SourceList.jsx:  const sources = useSelector(
static/js/components/SourceList.jsx:  const sourceTableEmpty = useSelector(
static/js/components/SuperAdminGroupList.jsx:  const groups = useSelector((state) => state.groups.all);
static/js/components/UserInfo.jsx:  const users = useSelector((state) => state.users);
```

Reply:
So, I guess at some point we got rid of the notion of containers vs components.  I don't have a problem with that.  Would be worth checking the redux docs to see their latest recommendation along with hooks.

Reply:
Yes, that was the major refactor when we adopted React hooks. Our pattern follows the latest recommendations outlined by the `react-redux` team: https://react-redux.js.org/next/api/hooks#useselector

Reply:
Dan Abramov:

> “ I don’t suggest splitting your components like this anymore. If you find it natural in your codebase, this pattern can be handy. But I’ve seen it enforced without any necessity and with almost dogmatic fervor far too many times. The main reason I found it useful was because it let me separate complex stateful logic from other aspects of the component. Hooks let me do the same thing without an arbitrary division.”

Reply:
Agreed 100% -- so glad we moved away from the presentational/container component split and just use hooks now!